trade-finance-manager-ui/server/types/view-models/utilisation-report-reconciliation-for-report-view-model.ts

```diff
@@ -109,9 +109,18 @@ export type PaymentDetailsRowViewModel = {
   };
 };
 
+export type SelectedFilter = { value: string; removeHref: string };
+
+export type SelectedPaymentDetailsFiltersViewModel = {
+  facilityId: SelectedFilter | null;
+  paymentCurrency: SelectedFilter | null;
+  paymentReference: SelectedFilter | null;
```

Comment:
Is it possible to do `?SelectedFilter` instead of `SelectedFilter | null`?

Reply:
I did this because on previous PRs you have preferred us to not explicitly set things to undefined and as writing it I realised I was explicitly setting undefined so used null instead. I guess with the above refactor I can switch to undefined by starting with `const mapped = {}` so I can change this.

Just to be clear for going forwards, is the preference to have fields be optional and then if the logic would require us to explicitly set to undefined to keep the code readable, then in that case the preference switches to not be optional but allow null instead?

Reply:
Ah sorry @RLavender98 so changing this would mean we have to specifically set `undefined` values? If so I think the above implementation (`| null`) is better 👍